crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -303,18 +313,25 @@ impl CallContractCommand {
 			self.proof_size = proof_size_input.parse::<u64>().ok(); // If blank or bad input, estimate it.
 		}
 
-		// Resolve who is calling the contract.
-		if self.suri == DEFAULT_URI {
-			// Prompt for uri.
-			self.suri = cli
-				.input("Signer calling the contract:")
-				.placeholder("//Alice")
-				.default_input("//Alice")
-				.interact()?;
-		};
+		// Resolve who is calling the contract. If a `suri` was provided via the command line, skip
+		// the prompt.
+		if self.suri == DEFAULT_URI && !self.use_wallet && message.mutates {
+			if cli.confirm("Do you want to use your browser wallet to sign the transaction? (Selecting 'No' will prompt you to manually enter the secret key URI for signing, e.g., '//Alice')")
```

Comment:
Assume this is the same block as found in the call chain implementation. We really should have a single function so we have a consistent experience. I suspect this may be due to feature gating, in which case the shared function can be located within the call module and feature gate to either contracts or parachains.

Reply:
https://github.com/r0gue-io/pop-cli/pull/371/commits/6035fd86b7ca679127b5cd34765a51dca6a66f13